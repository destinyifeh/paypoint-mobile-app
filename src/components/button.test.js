'use strict';

import React from 'react';
import Button from './button';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<Button />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('transparent button remains the same', () => {
  const tree = renderer
    .create(<Button
      transparent
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('disabled button remains the same', () => {
  const tree = renderer
    .create(<Button
      isDisabled
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('hidden button remains the same', () => {
  const tree = renderer
    .create(<Button
      hidden
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('loading button remains the same', () => {
  const tree = renderer
    .create(<Button
      loading
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
