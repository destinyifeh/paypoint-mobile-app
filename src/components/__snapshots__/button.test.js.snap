// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`disabled button remains the same 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#EE312A",
      "borderRadius": 25,
      "opacity": 1,
    }
  }
>
  <View
    accessible={true}
    clickable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderColor": "#2089dc",
          "borderRadius": 25,
          "borderWidth": 0,
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 12,
        }
      }
    />
  </View>
</View>
`;

exports[`hidden button remains the same 1`] = `<View />`;

exports[`loading button remains the same 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#EE312A",
      "borderRadius": 25,
      "opacity": 0.6,
    }
  }
>
  <View
    accessible={true}
    clickable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderColor": "#2089dc",
          "borderRadius": 25,
          "borderWidth": 0,
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 12,
        }
      }
    >
      <ActivityIndicator
        animating={true}
        color="white"
        hidesWhenStopped={true}
        size="small"
        style={
          Object {
            "marginVertical": 2,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#EE312A",
      "borderRadius": 25,
      "opacity": 1,
    }
  }
>
  <View
    accessible={true}
    clickable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderColor": "#2089dc",
          "borderRadius": 25,
          "borderWidth": 0,
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 12,
        }
      }
    />
  </View>
</View>
`;

exports[`transparent button remains the same 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "borderRadius": 25,
      "opacity": 1,
    }
  }
>
  <View
    accessible={true}
    clickable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderColor": "#2089dc",
          "borderRadius": 25,
          "borderWidth": 0,
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 12,
        }
      }
    />
  </View>
</View>
`;
